Set TCP_NODELAY on connections accepted by the layer (#4645)
metalbear-co/mirrord: changelog.d/+tcp-nodelay-accepted-sockets.changed.md

> Connections accepted on a mirrord-managed listener now have `TCP_NODELAY` set. Stolen and mirrored traffic is relayed to the local application one request at a time, so Nagle's algorithm could add a delayed-ACK stall to every request on servers that do not set the option themselves.

Reviewer: ```suggestion
Connections accepted by the layer now have `TCP_NODELAY` set to reduce latency.
```

Author: Done in 2d5ddb24 — applied verbatim.
